src/tridiag.jl

```diff
@@ -174,16 +174,13 @@ end
 isreal(S::SymTridiagonal) = isreal(S.dv) && isreal(S.ev)
 
 transpose(S::SymTridiagonal) = S
-adjoint(S::SymTridiagonal{<:Number}) = SymTridiagonal(vec(adjoint(S.dv)), vec(adjoint(S.ev)))
-adjoint(S::SymTridiagonal{<:Number, <:Base.ReshapedArray{<:Number,1,<:Adjoint}}) =
-    SymTridiagonal(adjoint(parent(S.dv)), adjoint(parent(S.ev)))
+adjoint(S::SymTridiagonal) = SymTridiagonal(_vecadjoint(S.dv), _vecadjoint(S.ev))
```

Comment:
I think this is only correct for `SymTridiagonal{<:Number}`.

In particular, the off-diagonal elements should be `transpose.(adjoint.(S.ev))` or equivalently `conj(S.ev)`.

(Why we allow `SymTridiagonal` for non-`Number` elements is beyond me.)